torch/optim/optimizer.py

```diff
@@ -164,16 +164,33 @@ def update_group(group, new_group):
             update_group(g, ng) for g, ng in zip(groups, saved_groups)]
         self.__setstate__({'state': state, 'param_groups': param_groups})
 
-    def zero_grad(self):
-        r"""Clears the gradients of all optimized :class:`torch.Tensor` s."""
+    def zero_grad(self, set_to_none=False):
+        r"""Set the gradients of all optimized :class:`torch.Tensor` s to zero.
+
+        Arguments:
+            set_to_none (bool): instead of setting to zero, set the grad to None.
+                This will in general have lower memory footprint, but using this
+                comes with caveats, to name a few:
```

Comment:
I wouldn't say these are caveats.  Some are benefits, like point 3 (skipping the update is faster if the grad is None).
Instead of
```
This will in general have lower memory footprint, but using this comes with caveats, to name a few:
```
I would say
```
This is will in general have lower memory footprint, and can modestly improve performance.  However, it changes certain behaviors.  For example:
```